redwood-treehouse/src/commonMain/kotlin/app/cash/redwood/treehouse/ZiplineTreehouseUi.kt

```diff
@@ -46,4 +49,12 @@ public interface ZiplineTreehouseUi : ZiplineService, EventSink {
   )
 
   public fun snapshotState(): StateSnapshot? = null
+
+  /** Access to the host that presents this UI. */
+  public interface Host : ZiplineService, ChangesSinkService {
+    public val uiConfigurations: StateFlow<UiConfiguration>
+    public val stateSnapshot: StateSnapshot?
+
+    public fun addOnBackPressedCallback(callback: OnBackPressedCallbackService): CancellableService
+  }
```

Comment:
Strictly speaking introducing this Host abstraction is a bit less efficient than passing the 4 parameters inline. This is because we need an extra 2 host-guest bridges for the two value parameters (uiConfigurations and stateSnapshot) that were originally passed by value into `start()`. I think this is a good tradeoff for simplicity.